src/libs/ReportUtils.ts

```diff
@@ -7077,9 +7084,15 @@ function shouldShowMerchantColumn(transactions: Transaction[]) {
 }
 
 /**
- * Whether the report is a system chat or concierge chat, depending on the user's account ID (used for A/B testing purposes).
+ * Whether the report is a system chat or concierge chat, depending on the onboarding report ID or fallbacking
+ * to the user's account ID (used for A/B testing purposes).
  */
 function isChatUsedForOnboarding(report: OnyxEntry<Report>): boolean {
+    // onboarding can be an array for old accounts and accounts created from olddot
+    if (!Array.isArray(onboarding) && onboarding?.chatReportID === report?.reportID) {
```

Comment:
To me it makes more sense to check what it is (i.e., object) instead of what it isn't
```suggestion
    if (onboarding && typeof onboarding === 'object' && onboarding?.chatReportID === report?.reportID) {
```

Reply:
@rojiphil you're right, if I try @chiragsalian suggestion I get a typecheck issue

Reply:
We're defining `onboarding` as `OnyxEntry<Onboarding | []>` above, so if we check if isn't an array, it will be an object of type `Onboarding`.

Reply:
oh,
```
onboarding = [];
typeof onboarding === 'object'
>> true
```
lol weird JS 😂 I wouldve thought that would return true for object and false for array. but i guess not. Cool looks like `!Array.isArray(onboarding)` is the correct approach here. Thanks gents.